server/internal/router/pod.go

```diff
@@ -0,0 +1,75 @@
+package router
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/llm-operator/inference-manager/server/internal/k8s"
+	apiv1 "k8s.io/api/core/v1"
+)
+
+// podEventHandler is the event handler for managing pod.
+type podEventHandler struct {
+	m     *routeMap
+	errCh chan error
+}
+
+var _ k8s.EventHandlerWithContext = &podEventHandler{}
+
+// newPodEventHandler creates a new PodEventHandler.
+func newPodEventHandler(
+	m *routeMap,
+	errCh chan error,
+) *podEventHandler {
+	return &podEventHandler{
+		m:     m,
+		errCh: errCh,
+	}
+}
+
+// ProcessAdd processed the add event.
+func (h *podEventHandler) ProcessAdd(ctx context.Context, obj interface{}) {
+	// o, ok := obj.(runtime.Object)
+	pod, ok := obj.(*apiv1.Pod)
+	if !ok {
+		h.errCh <- fmt.Errorf("cast runtime.Object type")
+		return
+	}
+	log.Printf("Create a pod, ip is %s\n", pod.Status.PodIP)
```

Comment:
```suggestion
	log.Printf("Creating a pod, ip is %s\n", pod.Status.PodIP)
```

Or maybe "Create" is a bit odd here since server itself is not creating a pod. Maybe

```suggestion
	log.Printf("Registering a pod (IP: %s) in the routing table\n", pod.Status.PodIP)
```